plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -242,8 +245,9 @@ private Multimap<Integer, FileStatus> getCommittedPartitions(URI committedAttemp
         }
     }
 
-    private URI getExchangeDirectory()
+    private URI getExchangeDirectory(int taskPartitionId)
     {
+        URI baseDirectory = baseDirectories.get(taskPartitionId % baseDirectories.size());
```

Comment:
Let's round-robin them

Reply:
Discussed offline, the current design requires a deterministic mapping from partitionId to bucket